File: backend/app/database/models/coach.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.database import Base


class Coach(Base):
    __tablename__ = "coachs"

    id_coach = Column(Integer, primary_key=True, autoincrement=True)
    nom = Column(String(100), nullable=False)
    prenom = Column(String(100), nullable=False)
    email = Column(String(150), nullable=False, unique=True, index=True)
    password_hash = Column(String(255), nullable=False)
    created_at = Column(DateTime, server_default=func.now())

    # Relations
    adherents = relationship("Adherent", back_populates="coach")
    seances_ems = relationship("SeanceEms", back_populates="coach")

File: flutter_app/lib/models/coach.dart
class Coach {
  final int id;
  final String nom;
  final String prenom;
  final String email;

  Coach({
    required this.id,
    required this.nom,
    required this.prenom,
    required this.email,
  });

  factory Coach.fromJson(Map<String, dynamic> json) {
    return Coach(
      id: json['id'] as int,
      nom: json['nom'] as String? ?? '',
      prenom: json['prenom'] as String? ?? '',
      email: json['email'] as String,
    );
  }
}